src/codewhisperer/util/dependencyGraph/goDependencyGraph.ts

```diff
@@ -0,0 +1,236 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { Uri } from 'vscode'
+import { DependencyGraph, DependencyGraphConstants, Truncation } from './dependencyGraph'
+import * as CodeWhispererConstants from '../../models/constants'
+import { existsSync, readdirSync, statSync } from 'fs'
+import { sleep } from '../../../shared/utilities/timeoutUtils'
+import { getLogger } from '../../../shared/logger'
+import path from 'path'
+import { readFileAsString } from '../../../shared/filesystemUtilities'
+
+const importRegex = /^\s*import\s+([^(]+?$|\([^)]+\))/gm
+const moduleRegex = /"[^"\r\n]+"/gm
+const packageRegex = /^package\s+(.+)/gm
+
+export class GoDependencyGraph extends DependencyGraph {
+    override async getSourceDependencies(uri: Uri, content: string): Promise<string[]> {
+        const imports = this.readImports(content)
+        const dependencies = this.getDependencies(uri, imports)
+        return dependencies
+    }
+
+    // Returns file paths of other .go files in the same directory declared with the same package statement.
+    override async getSamePackageFiles(uri: Uri): Promise<string[]> {
+        const fileList: string[] = []
+        const packagePath = path.dirname(uri.fsPath)
+        const fileName = path.basename(uri.fsPath)
+        const content = await readFileAsString(uri.fsPath)
+        const packageName = this.readPackageName(content)
+
+        const files = readdirSync(packagePath, { withFileTypes: true })
+        for (const file of files) {
+            if (file.isDirectory() || !file.name.endsWith(DependencyGraphConstants.goExt) || file.name === fileName) {
+                continue
+            }
+            const filePath = path.join(packagePath, file.name)
+            const content = await readFileAsString(filePath)
+            if (this.readPackageName(content) !== packageName) {
+                continue
+            }
+            fileList.push(filePath)
+        }
+
+        return fileList
+    }
+
+    override async isTestFile(content: string): Promise<boolean> {
+        const imports = this.readImports(content)
+        const filteredImports = imports.filter(importStr => {
+            return importStr.includes('"testing"')
+        })
+        return filteredImports.length > 0
+    }
+
+    override async generateTruncation(uri: Uri): Promise<Truncation> {
+        try {
+            const workspaceFolder = vscode.workspace.getWorkspaceFolder(uri)
+            if (workspaceFolder === undefined) {
+                this._pickedSourceFiles.add(uri.fsPath)
+            } else {
+                await this.searchDependency(uri)
+                await this.traverseDir(this.getProjectPath(uri))
+            }
+            await sleep(1000)
```

Comment:
Why sleep?

Reply:
Not entirely sure but all the other language dependency graphs call the same method here